kymatio/scattering1d/backend/tensorflow_backend.py

```diff
@@ -27,9 +27,14 @@ def subsample_fourier(cls, x, k):
         """
         cls.complex_check(x)
 
-        y = tf.reshape(x, (-1, k, x.shape[-1] // k))
+        shape = list(x.shape[:-1]) + [k, x.shape[-1]//k]
+        if shape[0] is None:
+            shape[0] = -1
```

Comment:
@janden @eickenberg do y'all think this additional code is fine? 

The Keras backend was erroring out when I tried to reshape. x.shape produces a datatype with None as the first element. The None specifies a general "batch" shape in keras. I need to convert this None to a -1 to play nice with Tensorflow.

Reply:
This sounds good to me!